src/sd/spi.rs

```diff
@@ -52,15 +59,16 @@ impl<'d> SpiDevice<'d> {
             gpio_wp_polarity: wp_polarity.unwrap_or(false), // Active when low
         };
 
-        Self {
+        Ok(Self {
             configuration,
-            driver,
+            _driver: driver,
+            host,
             _p: PhantomData,
-        }
+        })
     }
 
     pub fn get_host(&self) -> spi_host_device_t {
-        self.driver.host()
+        self.host
     }
 
     pub fn set_clock(&mut self, clock: u32) -> Result<(), EspError> {
```

Comment:
This is wrong, you are doing it on the wrong handle...

Reply:
Also, `enable_intrerrupt` and `wait_interrupt` are wrong for the same reasons.

Oh well... I think we need to more strictly follow the three layers in ESP IDF:
* SPI (or SDMMC) **host** driver; where the SPI driver also uses `SpiDriver)
* SDMMC **higher layer** driver (that uses one of the above host drivers)
* FAT driver (that uses the SDMMC higher layer driver from above)

Otherwise it is a bit messy, both for us, and for the users...